### DISCUSSION TITLE
Prompt with `{{CLIPBOARD}}` variable not working

### DISCUSSION THREAD

**Discussion by LiYanan2004** [open-webui/open-webui#7272] [Issues]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** 0.4.3
- **Ollama (if applicable):** 0.4.3

- **Operating System:** macOS 15.1
- **Browser (if applicable):** Safari(18.1 (20619.2.8.11.9)), Edge(131.0.2903.63)

## Expected Behavior:

Able to use custom prompt with {{CLIPBOARD}} variable

## Actual Behavior:

Got an error in console showing Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'navigator.clipboard.readText')

## Description

**Bug Summary:**
Prompt with {{CLIPBOARD}} variable not working

## Reproduction Details

**Steps to Reproduce:**

- Go to Workspace
- Add a prompt (with {{CLIPBOARD}} variable)
- Use that prompt in the new chat

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]
<img width="1030" alt="截屏2024-11-23 09 57 13" src="https://github.com/user-attachments/assets/92525f97-431d-422b-90d8-9352d080f72d">

**Docker Container Logs:**
Nope, everything seems great

**Screenshots/Screen Recordings (if applicable):**

https://github.com/user-attachments/assets/21eb50da-9622-4344-b6af-1a107d48187c

**Comment by silentoplayz** [comment]
I'm unable to reproduce on v0.4.4 with a slash command prompt containing all of Open WebUI's prompt variables.

**Reply by LiYanan2004** [reply]
@silentoplayz Try to add this prompt preset in workspace 

```
Refine the text below. Make sure just output refined version of text without additional narration.

{{CLIPBOARD}}
```

**Reply by silentoplayz** [reply]
Working great for me.
![image](https://github.com/user-attachments/assets/ba3fc81d-e967-4210-9d15-436a3c7fe92b)
![image](https://github.com/user-attachments/assets/b21494f7-21b8-4430-ad74-6b0ab7b7d59c)
![image](https://github.com/user-attachments/assets/62553ff9-7815-4d02-9a87-56ac39f1997f)
![image](https://github.com/user-attachments/assets/09e9d24d-ba58-41cf-95b1-73146b4e1973)

**Reply by LiYanan2004** [reply]
@silentoplayz Can you try on macOS environment? Seems like it’s working on Windows but not macOS (both edge and safari)

**Reply by silentoplayz** [reply]
I do not have a macOS environment to test this with.

**Reply by tjbck** [reply]
I tested this with MacOS and wasn't able to reproduce, hence why it's moved to issues here.

**Comment by tjbck** [comment]
Seems to be a secure context issue. Please refer to our docs.

**Reply by LiYanan2004** [reply]
I use it directly via the ip address of my desktop computer. You mean related to http?

**Comment by somera** [comment]
Hi, I have 3 prompts

![image](https://github.com/user-attachments/assets/ffaffce1-8346-4eaa-8dab-84673c131386)

"Review Code" prompt ist just text and I can use it.

"Review Bash Script" and "Review Python Script" uses ```{{CLIPBOARD}}```.
```
Please review the following Python script for readability, performance, and best practices. Identify any inefficiencies, security vulnerabilities, or areas that could be refactored for better maintainability. Suggest improvements and provide explanations where necessary.
{{CLIPBOARD}}
```
And I can't select and use both. Only the first one without ```{{CLIPBOARD}}``` works.

And I have no idea why. I don't see any errors.

My Open WebUI v0.5.7 run on a Linux system. I'm using it in WIndows with the current Chrome version.

**Comment by somera** [comment]
ok, I found this.

![image](https://github.com/user-attachments/assets/f4bd28a0-a00b-4c21-9b48-71928779ad6a)

What is the problem?

I have the content in the clipboard.

**Reply by LiYanan2004** [reply]
Check your url address bar, are you using http? If so, this will not working. Only works when using https

**Reply by somera** [reply]
> Check your url address bar, are you using http? If so, this will not working. Only works when using https

Hm ... yes, I'm using it with HTTP. It's my local network.

**Reply by krittaprot** [reply]
> Check your url address bar, are you using http? If so, this will not working. Only works when using https

@LiYanan2004 
I got the same issue when using http, the prompt template does not work...

How do I make it https? I tried but it https cannot load my openwebui server.

**Reply by somera** [reply]
To enable HTTPS on a website, you'll need an SSL/TLS certificate.

**Reply by athoik** [reply]
There is a section about https encryption in the [docs](https://docs.openwebui.com/getting-started/advanced-topics/https-encryption)

**Reply by krittaprot** [reply]
I found out this one works for me: https://pinggy.io/

Thanks @somera @athoik
